### ISSUE TITLE
oxlint standalone binary silently ignores jsPlugins — reports success on failing code

### ISSUE THREAD

**Issue by JumpLink** [oxc-project/oxc#25203]
### Summary

The standalone `oxlint` binary from the GitHub releases accepts a config containing `jsPlugins`, silently skips loading them, and reports no diagnostics from those plugins — with **no warning and no error**. When a project's violations come only from JS-plugin rules, the binary exits **0** where the npm launcher exits **1**.

### Reproduction

`x86_64-unknown-linux-gnu`, standalone `oxlint 1.76.0` (from `apps_v1.76.0`) vs npm `oxlint 1.72.0`.

`.oxlintrc.json`
```json
{ "jsPlugins": ["./plug/index.js"], "rules": { "demo/no-forbidden": "error" } }
```

`plug/index.js`
```js
export default {
  meta: { name: 'demo' },
  rules: {
    'no-forbidden': {
      create(ctx) {
        return {
          Identifier(node) {
            if (node.name === 'forbidden') ctx.report({ node, message: 'JS-PLUGIN RULE FIRED' });
          },
        };
      },
    },
  },
};
```

`only.js`
```js
const forbidden = 1;
export { forbidden };
```

| | diagnostics | exit |
|---|---|---|
| `node node_modules/oxlint/bin/oxlint only.js` | 3 errors | 1 |
| `./oxlint-x86_64-unknown-linux-gnu only.js` | *(no output)* | **0** |

Adding a rule the Rust linter does know (`no-debugger`) makes the difference less obvious but no less real — 4 diagnostics vs 1, both exiting 1. The exit-code divergence above is the case that matters: a CI using the standalone binary goes green on code that violates the project's own rules.

Also reproduced on `aarch64-unknown-linux-musl` (postmarketOS, no Node installed), which is how we ran into it — the standalone binary is the only way to lint on a host without Node.

### Why this looks unintended rather than a documented limitation

The correct diagnostic already exists and is simply unreachable.

`ConfigBuilderError::NoExternalLinterConfigured` (`crates/oxc_linter/src/config/config_builder.rs:802`) reads:

> `` `jsPlugins` config contains '{plugin_specifier}'. JS plugins are not supported on 32-bit or big-endian platforms at present. ``

but it is constructed inside

```rust
if !external_plugins.is_empty() && external_plugin_store.is_enabled() {
```

(`config_builder.rs:250`). In the standalone binary `is_enabled` is `false`: `apps/oxlint/src/main.rs:43` constructs `CliRunner::new(command, None)`, which reaches `ExternalPluginStore::new(self.external_linter.is_some())` at `apps/oxlint/src/lint.rs:223`. The whole block is skipped before the error can fire, so a config naming a plugin is accepted and the plugin is dropped without a word.

The `is_enabled` gate appears to come from 6e8d2f6318 *"fix(language_server): ignore JS plugins (#14379)"*, where silence is right for the LSP; the standalone binary inherited it.

Verified byte-identical on `main` today (`apps/oxlint/src/main.rs`, still `None`, still no warning).

Neither the [plugins](https://oxc.rs/docs/guide/usage/linter/plugins.html) nor the [js-plugins](https://oxc.rs/docs/guide/usage/linter/js-plugins.html) page mentions that JS plugins require the Node-based distribution. Related: oxc-project/website#1125 (open since May) asks for exactly such a note on the pure-Rust CLI.

### Suggested fix

Not asking for JS-plugin support in the pure-Rust binary — that needs an embedded JS engine, and #22634 already settled the equivalent request for oxfmt. Only that the skip be **observable**:

- a stderr warning at `config_builder.rs:250` naming the specifier when `!external_plugins.is_empty() && !external_plugin_store.is_enabled()`, e.g. *"`jsPlugins` are configured but this build cannot run them; rules from these plugins are skipped. Use the npm `oxlint` package (requires Node.js) to run them."*
- Backwards compatible by construction: a config without `jsPlugins` fails the first condition, so output and exit code are byte-for-byte unchanged for everyone else.
- Optionally `--no-js-plugins` to silence it for people deliberately running the Rust CLI against a shared config.

A hard error (reusing the existing variant, reworded to cover standalone builds alongside 32-bit/big-endian) would be more correct but changes exit codes for current users — that call is yours. Happy to open a PR for whichever you prefer.

Sibling silent-drop, still open: #22117 (`--print-config` drops `jsPlugins` rules from its output).

### Disclosure

This report was prepared with AI assistance (Claude). The AI performed the source analysis and ran the reproductions; every code reference and both measurements above were verified against the actual binaries and the checked-out source before submission. The framing, the decision to file, and the final text were reviewed by me.
